components/app/manage/creator/TextField.vue

```diff
@@ -0,0 +1,82 @@
+<template>
+  <form-control
+    :value="value"
+    :label="required ? `* ${label}` : label"
+    :hint="hint"
+    :max-length="maxLength"
+  >
+    <input
+      :value="value"
+      :maxlength="maxLength"
+      :placeholder="placeholder"
+      type="text"
+      class="input input-primary bg-base-200 text-base-content"
+      @blur="validate($event.target.value)"
+      @input="
+        validate($event.target.value);
+        $emit('input', $event.target.value);
+      "
+    />
+
+    <template #hint>
+      <span class="label-text-alt text-error">
+        {{ details }}
+      </span>
+    </template>
+  </form-control>
+</template>
+
+<script>
+import FormControl from '@/components/app/manage/creator/FormControl';
+
+export default {
+  name: 'TextField',
+
+  components: { FormControl },
+
+  props: {
+    value: {
+      type: String,
+      default: '',
+    },
+    required: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+    label: {
+      type: String,
+      required: true,
+    },
+    placeholder: {
+      type: String,
+      required: true,
+    },
+    hint: {
+      type: String,
+      required: false,
+      default: null,
+    },
+    maxLength: {
+      type: Number,
+      required: false,
+      default: 0,
+    },
+  },
+
+  data() {
+    return {
+      details: null,
+    };
+  },
+
+  methods: {
+    validate(val) {
```

Comment:
suggestion(non-blocking): maybe turn this into a function, I've seen the same implementation at some other places as well.

Reply:
Keeping this for now since we have it only at two places. We should consider using a dedicated library for validation in the future. See [vue-validate](https://vee-validate.logaretm.com/v4/).